packages/eas-cli/src/utils/queries.ts

```diff
@@ -131,3 +127,70 @@ async function paginatedQueryWithSelectPromptInternalAsync<
     items => promptOptions.getIdentifierForQueryItem(items) === valueOfUserSelectedListItem
   );
 }
+
+export async function paginatedQueryWithMultiSelectPromptAsync<
+  QueryReturnType extends Record<string, any>
+>(
+  queryArgs: PaginatedQueryWithSelectPromptArgs<QueryReturnType>
+): Promise<QueryReturnType[] | void> {
+  return await paginatedQueryWithMultiSelectPromptInternalAsync(queryArgs, [], []);
+}
+
+async function paginatedQueryWithMultiSelectPromptInternalAsync<
+  QueryReturnType extends Record<string, any>
+>(
+  {
+    limit,
+    offset,
+    queryToPerform,
+    promptOptions,
+  }: PaginatedQueryWithSelectPromptArgs<QueryReturnType>,
+  queryItemAccumulator: QueryReturnType[],
+  selectedIdsAccumulator: string[]
+): Promise<QueryReturnType[] | void> {
+  // query an extra item to determine if there are more pages left
+  const paginatedItems = await queryToPerform(limit + 1, offset);
+  const areMorePagesAvailable = paginatedItems.length > limit;
+  // drop that extra item used for pagination from our render logic
+  const currentPage = paginatedItems.slice(0, limit);
+  const newQueryItemAccumulator = [...queryItemAccumulator, ...currentPage];
+
+  const selectionPromptListItems = uniqBy(newQueryItemAccumulator, queryItem =>
+    promptOptions.getIdentifierForQueryItem(queryItem)
+  ).map<ExpoChoice<string>>(queryItem => ({
+    title: promptOptions.createDisplayTextForSelectionPromptListItem(queryItem),
+    value: promptOptions.getIdentifierForQueryItem(queryItem),
+    selected: selectedIdsAccumulator.includes(promptOptions.getIdentifierForQueryItem(queryItem)),
+  }));
+  if (areMorePagesAvailable) {
+    selectionPromptListItems.push({ title: 'Next page...', value: fetchMoreValue });
+  }
+  if (selectionPromptListItems.length === 0) {
+    return;
+  }
+
+  const valueOfUserSelectedListItems = await multiselectAsync<string>(
```

Comment:
It's hard to see from the recording, but I understand that the way it works to switch to the next page you need to press space to mark it and enter to submit, which displays new prompt with a next page.

I don't think this is a good experience, I have 2 alternatives:
- When the next page is marked with space immediately go to the next page. All items that were already checked should also show up plus entire new page of unchecked items(This assumes that list displayed to users might grow).
- When the next page is marked with space immediately go to the next page, but on every page after the first one, also display the previous page entry.

I think the first approach would be cleaner from users perspective.

As for how to implement that, I think `onState` callback should provide that info. And if it's not possible with `prompts`, maybe consider different lib for multiselect.

Reply:
I also thought: "this doesn't seem ideal"! But it seemed the best we could do. Thanks for the alternative approaches. I'll investigate to see if I can cook up something that feels a bit better to use.

Reply:
I tried to test that, but it seems you are not using that code in any of the commands, can you add it to the intended command or update the recording?

Reply:
In general, if the alternative implementation is a problem I would be leaning into keeping old implementation and setting a default limit to sth most people would never hit.

Reply:
Sorry, I forgot to record a video. I think the UX of this approach is what we were hoping for. However, the library is untyped and subclassing to get the logic we want seems unnecessarily verbose and maybe a little fragile. Added a video to the description.

However, I think this is better than setting a super high limit. We've seen hundreds on apps/accounts and this will soon lead to timeouts for those users when they try to run these commands.

Reply:
> I think setting a super high limit for device queries is the wrong approach.

I did not suggest increasing that to huge values, just to sth that the average user would not hit. I did not realize that the default limit is already at 50, if that is the case I'm fine with either approach

> However, the library is untyped and subclassing to get the logic we want seems unnecessarily verbose and maybe a little fragile.

I don't think this is a problem, I was more concerned about size with adding one more library, but it seems it did not affect much.

Reply:
In terms of UX it's not exactly what I had in mind. You are currently creating a new prompt for every page, that is left in the terminal. Ideally, that logic should be inside multi-select, so when you fetch the next page state of the existing prompt is updated instead, but I'm not sure how possible it is with this lib.

Reply:
cc @brentvatne any opinion on that?

Reply:
(i followed up with kirby and suggested skipping this functionality)